libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -167,7 +168,31 @@ public String getStoredOrgId() {
 	 * @return Current user that's logged in.
 	 */
 	public UserAccount getCurrentUser() {
-		return buildUserAccount(getCurrentAccount());
+		cachedCurrentUserAccount = buildUserAccount(getCurrentAccount());
+		return cachedCurrentUserAccount;
+	}
+
+	/**
+	 * Returns a cached value of the current user.
+	 *
+	 * NB: The oauth tokens might be outdated
+	 *     Should be used by methods that only care about the current user's identity (org id, user id etc)
+	 *     Is faster than getCurrentUser()
+	 *
+	 * @return Current user that's logged in (with potentially outdated oauth tokens)
+	 */
+	public UserAccount getCachedCurrentUser() {
```

Comment:
Should we maintain this at this layer, or move this to the layer that needs it, i.e. plugins? If we provide this here, we will need to keep things in sync for any user account state changes and all components will be affected.

Reply:
I thought about doing that at first, but it would mean keeping (or getting) a reference to the SmartStorePlugin from SalesforceHybridSDKManager.cleanUp(UserAccount ua). There is no clean way to do that.

Also we would have to do the same work in SalesforceReactSDKManager.cleanUp(UserAccount xx) for SmartStoreReactBridge. So it would be duplicated code.

Reply:
All you need is a broadcast receiver for this.

Reply:
I need a Context to register the broadcast receiver. Also isn't asynchronous, couldn't a call to getCurrentUser() pick up the cached value when it shouldn't?